core/src/main/java/dk/alexandra/fresco/lib/lp/LPTableau.java

```diff
@@ -45,10 +45,16 @@ public LPTableau(Matrix<DRes<SInt>> C, ArrayList<DRes<SInt>> B,
     return z;
   }
 
-  public void toString(ProtocolBuilderNumeric builder) {
-    builder.debug().openAndPrint("C: ", C, System.out);
-    builder.debug().openAndPrint("B: ", B, System.out);
-    builder.debug().openAndPrint("F: ", F, System.out);
-    builder.debug().openAndPrint("z: ", z, System.out);
+  /**
+   * Opens and outputs the plaintext values of this tableau for debugging.
+   * 
+   * @param builder a builder to provide the open and print functionality
+   * @param ps a PrintStream on which to print the debugging information
+   */
+  public void debugInfo(ProtocolBuilderNumeric builder, PrintStream ps) {
+    builder.debug().openAndPrint("C: ", C, ps);
+    builder.debug().openAndPrint("B: ", B, ps);
+    builder.debug().openAndPrint("F: ", F, ps);
+    builder.debug().openAndPrint("z: ", z, ps);
```

Comment:
At this point it would be nice to be able to reconfigure the output, so it is usable in the test - the test should inspect the debugged variables. However this is not possible with the current implementation of debug

Reply:
I am not exactly sure what you are thinking here.

Reply:
Basically your tests are working from the design that the performance logger outputs strings. This entails testing on the form of String.contains("KeyOfMyIndex") and String.contains(String.valueOf(myKeyValue)) 

A more reasonable design in this context exposes the performance counters more directly. My suggestion is in this branch to alter the tests such that they introduce a helper method take the performance result (the string), a key ("z") and a value (z) and tests it currently via the string, but is easily reconfigured should the logger interface receive a bit of lova and change